lib/src/parsed_path.dart

```diff
@@ -167,7 +167,8 @@ class ParsedPath {
   /// Returns a two-element list. The first is the name of the file without any
   /// extension. The second is the extension or "" if it has none.
   List<String> _splitExtension() {
-    final file = parts.lastWhere((p) => p != '', orElse: () => null);
+    final file =
+        parts.cast<String?>().lastWhere((p) => p != '', orElse: () => null);
```

Comment:
This was REALLY tricky to figure out! It'd be great if we could offer help here!